### PR TITLE
Bug 1203551 - Don't allow count to exceed MAX_JOBS_COUNT in the jobs api

### PR DESCRIPTION
**Pull request by KWierso**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1594)

### CODE REVIEW

**Comment by edmorley** [comment]
I think for this and the other views we should use the django-rest-framework pagination feature rather than our own? ie:
http://www.django-rest-framework.org/api-guide/pagination/#limitoffsetpagination

This is actually used already by a couple of endpoints, but we should probably enable it by default in settings.py and override where needed:
http://www.django-rest-framework.org/api-guide/pagination/#modifying-the-pagination-style

Thoughts @wlach @camd ?

**Comment by edmorley** [comment]
(We could add the backwards compatibility with the `count` parameter by subclassing `rest_framework.pagination.LimitOffsetPagination` and using that until people have switched over to `limit` instead of `count`. Or just leave in indefinitely.)

**Comment by camd** [comment]
Ed, yeah, I totally agree.  I would much rather use something standard to DRF.

File: treeherder/webapp/api/jobs.py [hunk 1]
```diff
@@ -88,10 +89,15 @@ def list(self, request, project, jm):
...
         """
+        MAX_JOBS_COUNT = 2000
         filter = UrlQueryFilter(request.query_params)
 
         offset = int(filter.pop("offset", 0))
-        count = min(int(filter.pop("count", 10)), 2000)
+        count = int(filter.pop("count", 10))
+
+        if count > MAX_JOBS_COUNT:
+            return Response({"error": "Specified count exceeds API MAX_JOBS_COUNT value: " + str(MAX_JOBS_COUNT)}, status=status.HTTP_400_BAD_REQUEST)
```
Comment (camd) [inline treeherder/webapp/api/jobs.py]: concatenating values to strings is done with ``format()``, and it's best to wrap to keep things as close to 80 chars wide as possible like this:

    return Response(
        {"error": "Specified count exceeds API MAX_JOBS_COUNT value: {}".format(MAX_JOBS_COUNT)},
        status=status.HTTP_400_BAD_REQUEST
    )

**Comment by edmorley** [comment]
Since my comment it occurred to me that these two endpoints are still using the jobs model/datasource, so in fact we cannot use django-rest-framework's own pagination.

As such, let's land these like this for now, and when we finally get rid of datasource, we can switch to the d-r-f paginator :-)

**Pull request merged**